store/tikv/region_request.go

```diff
@@ -263,22 +270,74 @@ func (s *RegionRequestSender) SendReqCtx(
 	}
 }
 
-func (s *RegionRequestSender) sendReqToRegion(bo *Backoffer, ctx *RPCContext, req *tikvrpc.Request, timeout time.Duration) (resp *tikvrpc.Response, retry bool, err error) {
-	if e := tikvrpc.SetContext(req, ctx.Meta, ctx.Peer); e != nil {
+// RPCCancellerCtxKey is context key attach rpc send cancelFunc collector to ctx.
+type RPCCancellerCtxKey struct{}
+
+// RPCCanceller is rpc send cancelFunc collector.
+type RPCCanceller struct {
+	sync.Mutex
+	allocID   int
+	cancels   map[int]func()
```

Comment:
Is it necessary to use `map`? because it seems here just one cancel function.

Reply:
one CopIterator will fork many CopWorkers and parallel send CopTasks, so rpcCancel will catch more than one CancelFunc, and at last, we need use map instead of slice to make remove itself easier when cancel be called

Reply:
I think only a cancel function in copIterator is ok, becase copIteratorWorker can share the cancel to stop work.